ci_scripts/assert_news.py

```diff
@@ -90,49 +90,71 @@ def find_news_file(self):
         self.git.checkout(self.current_branch)
         self.git.set_upstream_branch(self.current_branch)
         self.git.pull()
-        current_commit_hash = self.git.get_commit_hash()
-        # Delete `master` as it may already exist and be set to the one of interest
+        current_commit = self.git.get_current_commit()
+        # Delete `master` as it may already exist and be set to the one of interest (on CircleCI anyway and maybe on other systems)
         if self.git.branch_exists(self.master_branch):
             self.git.delete_branch(self.master_branch)
         self.git.checkout(self.master_branch)
         self.git.set_upstream_branch(self.master_branch)
         self.git.pull()
-        master_branch_commit_hash = self.git.get_commit_hash()
+        master_branch_commit = self.git.get_current_commit()
         self.git.checkout(self.current_branch)
         added_news = self.git.get_changes_list(
-            'a', self.git.get_branch_point(
-                master_branch_commit_hash, current_commit_hash),
-            current_commit_hash,
-            os.path.relpath(NEWS_DIR,
-                            os.path.commonprefix([NEWS_DIR, PROJECT_ROOT]))
+            self.git.get_branch_point(
+                master_branch_commit, current_commit),
+            current_commit, change_type='a',
+            dir=os.path.relpath(NEWS_DIR,
+                                os.path.commonprefix([NEWS_DIR, PROJECT_ROOT]))
         )
         extension_to_exclude = ['.toml', '.rst']
         return [path for path in added_news if
                 len([ex for ex in extension_to_exclude if ex in path]) == 0]
 
+    def is_feature_branch(self):
+        """States whether the current branch contains new features.
+
+        Returns:
+            True if current branch is a feature branch; False otherwise.
+        """
+        return not self.is_special_branch()
+
+    def is_special_branch(self):
+        """Checks if current branch is special.
+
+        States whether the current branch is special as it may be used
```

Comment:
This docstring is a bit redundant. If you take function name into consideration, we're told three times that there's a "is branch special" check performed, but I still don't know what _special_ means. Also it's not immediately clear to me what "as it may be used for the release process" means.